native/src/routeSegment.h

```diff
@@ -107,27 +108,41 @@ struct RouteSegment {
 };
 
 struct RouteSegmentPoint : RouteSegment {
-public:
-    RouteSegmentPoint(SHARED_PTR<RouteDataObject>& road, int segmentStart) :
-				RouteSegment(road, segmentStart) {					
-                }
-    ~RouteSegmentPoint(){
-    }
-    double dist;
-    int preciseX;
-    int preciseY;
-    vector< SHARED_PTR<RouteSegmentPoint> > others;
+   public:
+	RouteSegmentPoint(SHARED_PTR<RouteDataObject>& road, int segmentStart) : RouteSegment(road, segmentStart) {}
+	~RouteSegmentPoint() {}
+	double dist;
+	int preciseX;
+	int preciseY;
+	vector<SHARED_PTR<RouteSegmentPoint>> others;
+
+	SHARED_PTR<LatLon> getPreciseLatLon() {
+		return std::make_shared<LatLon>(get31LatitudeY(preciseY), get31LongitudeX(preciseX));
+	}
 };
 
 struct FinalRouteSegment : RouteSegment {
-public:
-    FinalRouteSegment(SHARED_PTR<RouteDataObject>& road, int segmentStart) : RouteSegment(road, segmentStart) {
-    }
-        
-    ~FinalRouteSegment() {
-    }
-    bool reverseWaySearch;
-    SHARED_PTR<RouteSegment> opposite;
+   public:
+	FinalRouteSegment(SHARED_PTR<RouteDataObject>& road, int segmentStart) : RouteSegment(road, segmentStart) {}
+
+	~FinalRouteSegment() {}
+	bool reverseWaySearch;
+	SHARED_PTR<RouteSegment> opposite;
+};
+
+struct GpxPoint {
+	int32_t ind;
+	double lat;
+	double lon;
+	double cumDist;
+	SHARED_PTR<RouteSegmentPoint> pnt;
+	vector<SHARED_PTR<RouteSegmentResult>> routeToTarget;
+	vector<SHARED_PTR<RouteSegmentResult>> stepBackRoute;
+	int targetInd = -1;
+	bool straightLine = false;
+
+	GpxPoint(int32_t ind, double lat, double lon, double cumDist)
+		: ind(ind), lat(lat), lon(lon), cumDist(cumDist), pnt(nullptr), routeToTarget(0), stepBackRoute(0){};
```

Comment:
shared_ptr initialize nullptr by default, std::vector is init to empty by default